.buildkite/pipeline.yml

```diff
@@ -25,3 +25,50 @@ steps:
       - build-apple
     artifact_paths:
       - "screenshots/**/*.png"
+
+  - label: ":bazel: build"
+    key: bazel-build
+    command:
+      [
+        "bazelisk",
+        "build",
+        "--config=buildkite",
+        "--",
+        "//FlappyKite:FlappyKite",
+      ]
+    cache:
+      paths:
+        - ".cache"
+      size: "20g"
+    plugins:
+      - docker-compose#v5.3.0:
+          run: bazel-remote
+          config: .buildkite/docker-compose.yml
+          shell: false
+    agents:
+      queue: hosted_linux_small
+
+  - label: ":bazel: test"
+    key: bazel-test
+    depends_on:
+      - bazel-build
+    command:
+      [
+        "bazelisk",
+        "build",
+        "--config=buildkite",
+        "--",
+        "//FlappyKiteTests:FlappyKiteTests",
+        "//FlappyKiteUITests:FlappyKiteUITests",
+      ]
+    cache:
+      paths:
+        - ".cache"
+      size: "20g"
```

Comment:
I'd suggest sharing a cache specifically between the `bazel-test` and `bazel-build` steps. You can add the `name` to the cache in both steps. This means you'll only use that cache in these two steps.

```
cache:
      paths:
        - ".cache"
      size: "20g"
      name: "bazel"
```